privacyidea/lib/utils.py

```diff
@@ -849,11 +849,19 @@ def hash_password(password, hashtype):
         return "{SSHA512}" + pw
     elif hashtype == "OTRS":
         password = hashlib.sha256(password).hexdigest()
+    elif hashtype == "MD5CRYPT":
+        salt = geturandom(4, True)
+        password = crypt.crypt(password, "$1$" + salt + "$")
+    elif hashtype == "SHA512CRYPT":
+        salt = geturandom(4, True)
+        password = crypt.crypt(password, "$6$" + salt + "$")
```

Comment:
To be able to choose these hashtypes (MD5CRYPT, SHA512CRYPT) in a configurable SQL Resolver, you need to add this to the WebUI dropdown here:

https://github.com/privacyidea/privacyidea/blob/master/privacyidea/static/components/config/controllers/configControllers.js#L990